implementations/elixir/build.gradle

```diff
@@ -85,7 +85,7 @@ apps.each { app ->
     dependsOn checkLocalHexAndRebar
     doLast {
       exec {
-        commandLine mix('do deps.get, test')
+        commandLine mix('do deps.get, test --max-cases=1')
```

Comment:
Is it worth it to try with just the port numbers changes and the rest of the files in this PR first before introducing this one? I'd usually try to localize which test modules use some kind of global state that can be race-y and mark them `async: false` with a doc comment about why.

Does gradle itself parallelize the different sub-packages' test suites at all? If not, I'm not sure how we would get collisions across separate test suites, and when I spot checked the changed modules here, already none of them are using `use ExUnit.Case, async: true` so they shouldn't run parallel to each other within the suite either.

Reply:
I don't think the main issue is with `async: true` tests, rather different test cases impacting each other.
Gradle runs different app tests in different containers (in parallel), so the only overlap is between different test modules in the same app.

Reply:
I too have trouble understanding what is the reason for setting `max_cases` to 1. Tests within one module will still be executed the same way as before. AFAIK `max_cases` only impacts `async: true` tests

Reply:
> AFAIK max_cases only impacts async: true tests (i.e. how many async: true test modules run in parallel)

According to the `mix test` docs, it runs different modules in parallel `--max-cases - sets the maximum number of tests running asynchronously. Only tests from different modules run in parallel. Defaults to twice the number of cores`

Which is what we've seen. We don't use `async: :true` in the tests which had clashes.

Reply:
That's an interesting point. The tests where port clash happened did use `spawn_supervised` and did not use `async: true` but still got port clash.
If that's not because tests running in parallel, I'm not sure how that's possible.
I will roll back `--max-cases` for now.

Reply:
> That's an interesting point. The tests where port clash happened did use `spawn_supervised` and did not use `async: true` but still got port clash. If that's not because tests running in parallel, I'm not sure how that's possible.

In this test the listener is not supervised: 
https://github.com/build-trust/ockam/blob/060ab64adeac8109c9f4d0911ebff90865c7b2fc/implementations/elixir/ockam/ockam_kafka/test/interceptor/interceptor_test.exs#L127-L131
And I'm pretty sure it not terminating before we called to set inlets in this test https://github.com/build-trust/ockam/blob/060ab64adeac8109c9f4d0911ebff90865c7b2fc/implementations/elixir/ockam/ockam_kafka/test/interceptor/inlet_manager_test.exs#L43 was the culprit